src/exceptions.jl

```diff
@@ -28,6 +93,295 @@ end
+end
+
+"""
+    ObjectLostError <: RayError
+
+Indicates that the object is lost from distributed memory, due to node failure or system
+error.
+"""
+struct ObjectLostError <: RayError
+    object_context::ObjectContext
+end
+
+function Base.showerror(io::IO, ex::ObjectLostError)
+    print(io, "$ObjectLostError: ")
+    print_object_lost(io, ex.object_context)
+    print(io, m"""
+              All copies of $(ex.object_context.object_ref_hex) have been lost due to node
+              failure. Check cluster logs ("/tmp/ray/session_latest/logs") for more
+              information about the failure.
+              """)
+
+    return nothing
+end
+
+"""
+    ObjectReconstructionFailedError <: RayError
+
+Indicates that the object cannot be reconstructed.
+"""
+struct ObjectReconstructionFailedError <: RayError
+    object_context::ObjectContext
+end
+
+function Base.showerror(io::IO, ex::ObjectReconstructionFailedError)
+    print(io, "$ObjectReconstructionFailedError: ")
+    print_object_lost(io, ex.object_context)
+    print(io, m"""
+              The object cannot be reconstructed because it was created by an actor, a
+              `Ray.put` call, or its `ObjectRef` was created by a different worker.
+              """)
+    return nothing
+end
+
+"""
+    ObjectReconstructionFailedLineageEvictedError <: RayError
+
+Indicates that the object cannot be reconstructed because its lineage was evicted due to
+memory pressure.
+"""
+struct ObjectReconstructionFailedLineageEvictedError <: RayError
+    object_context::ObjectContext
+end
+
+function Base.showerror(io::IO, ex::ObjectReconstructionFailedLineageEvictedError)
+    print(io, "$ObjectReconstructionFailedLineageEvictedError: ")
+    print_object_lost(io, ex.object_context)
+    print(io, m"""
+              The object cannot be reconstructed because its lineage has been evicted to
+              reduce memory pressure. To prevent this error, set the environment variable
+              RAY_max_lineage_bytes=<bytes> (default 1GB) during `ray start`.
+              """)
+    return nothing
+end
+
+"""
+    ObjectReconstructionFailedMaxAttemptsExceededError <: RayError
+
+Indicates that the object cannot be reconstructed because the maximum number of task retries
+has been exceeded.
+"""
+struct ObjectReconstructionFailedMaxAttemptsExceededError <: RayError
+    object_context::ObjectContext
+end
+
+function Base.showerror(io::IO, ex::ObjectReconstructionFailedMaxAttemptsExceededError)
+    print(io, "$ObjectReconstructionFailedMaxAttemptsExceededError: ")
+    print_object_lost(io, ex.object_context)
+
+    # TODO: Update this message with more details on how to set `max_retries` once
+    # implemented: https://github.com/ray-project/ray/blob/ray-2.5.1/python/ray/exceptions.py#L593
+    print(io, m"""
+              The object cannot be reconstructed because the maximum number of task retries
+              has been exceeded. To prevent this error, set `max_retries` (default 3).
```

Comment:
I'm not sure this is accurate; with #215 I think we default to 0? https://github.com/beacon-biosignals/Ray.jl/blob/6e11dec0bcdbbfbf5286a4929d22280334dc6307/src/runtime.jl#L144

Reply:
Good catch. I wrote this code I believe before we implemented retries